python/TestHarness/testers/RunApp.py

```diff
@@ -108,6 +108,9 @@ def getCommand(self, options):
       # also, neither is the conflicting option "warn-unused"
       specs['cli_args'].append('--error-unused')
 
+    if '--check-input' in specs['cli_args']:
+      specs['status_message'] = 'SYNTAX_PASS'
```

Comment:
This shouldn't be an input parameter, instead we should have a method on tester for retrieving the correct status message.